models/articles.models.js

```diff
@@ -14,20 +14,49 @@ exports.fetchArticleById = (id) => {
         return article;
         })
 }; 
-exports.fetchArticles = () => {
+exports.fetchArticles = (userQuery = 'created_at', userOrder = 'DESC', userTopic, topics) => {
+  
+const lowerCaseQuery = userQuery.toLowerCase()
+const upperCaseOrder = userOrder.toUpperCase()
+const validQuery = ['article_id', 'title', 'topic', 'author', 'body', 'created_at', 'votes'];
+const validOrder = ['ASC', 'DESC']
+const validTopics = topics.map((topic) => {
+      return topic.slug
+})
+
+if(!validTopics.includes(userTopic) && userTopic) {
+  return Promise.reject({status: 404, msg: 'article not found'})
+}
+
+if(validTopics.includes(userTopic)) {
+  let optionalTopic = `WHERE articles.topic = $1`
+
+if(validQuery.includes(lowerCaseQuery) && validOrder.includes(upperCaseOrder)) {
+
+   return db
+   .query(`SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.body, articles.created_at, articles.votes, COUNT(comments.comment_id) AS comment_count
+   FROM articles 
+   LEFT JOIN comments ON comments.article_id = articles.article_id
+   ${optionalTopic}
+   GROUP BY articles.article_id
+   ORDER BY ${lowerCaseQuery } ${upperCaseOrder};`, [userTopic])
+   .then(({rows: articles})=> {
+     return articles;
+     })
+    } }
+ 
+else (validQuery.includes(lowerCaseQuery) && validOrder.includes(upperCaseOrder)) 
+ 
     return db
```

Comment:
rather than having the query written out twice, you can just += your query string if a topic exists or if there is a valid sort_by or order_by. This will make your code more DRY